state/protocol/badger/validity_test.go

```diff
@@ -145,3 +146,53 @@ func TestEntityExpirySnapshotValidation(t *testing.T) {
 		require.NoError(t, err)
 	})
 }
+
+func TestValidateVersionBeacon(t *testing.T) {
+	t.Run("no version beacon is ok", func(t *testing.T) {
+		snap := new(mock.Snapshot)
+
+		snap.On("VersionBeacon").Return(nil, nil)
+
+		err := validateVersionBeacon(snap)
+		require.NoError(t, err)
+	})
+	t.Run("height must be below highest block", func(t *testing.T) {
+		snap := new(mock.Snapshot)
+		block := unittest.BlockFixture()
+		block.Header.Height = 12
+
+		vb := &flow.SealedVersionBeacon{
+			SealHeight: uint64(37),
```

Comment:
We should add a valid version beacon here, so that the _only_ thing invalid about the input is the thing we want to test. Otherwise the test might pass because validation fails somewhere else, even if we don't actually validate the seal height properly.